src/MusicAnnouncementsMod.cs

```diff
@@ -50,6 +51,11 @@ private void RainWorld_OnModsInitHK(On.RainWorld.orig_OnModsInit orig, RainWorld
 		private void SongHK(On.Music.Song.orig_ctor orig, Song self, MusicPlayer musicPlayer, string name, MusicPlayer.MusicContext context)
 		{
 			orig(self, musicPlayer, name, context);
+			if (!MusicAnnouncementsConfig.ingameText.Value) // Gameplay announcement disabled
+			{
+				Debug.Log("(MusicAnnouncements) Skipping gameplay announcement due to config");
+				return;
+			}
 			if (context != MusicPlayer.MusicContext.StoryMode) // Ingame music only.
```

Comment:
This works, but it also has the side effect of disabling the pause menu text, since `songToAnnounce` is set below this and the menu text requires that in order to be created:
https://github.com/SabreML/MusicAnnouncements/blob/820c03412d7cd5f09120e0e86d29d77037553733/src/MusicAnnouncements_PauseMenu.cs#L21-L29
<hr>

A better spot for this would be right at the end of the method, edited slightly so that it only sets `announceAttempts` if the config setting is enabled:
(This way `songToAnnounce` is still set for the pause menu, and no attempts are made to announce it in-game.)
```suggestion
			orig(self, musicPlayer, name, context);
			if (context != MusicPlayer.MusicContext.StoryMode) // Ingame music only.
```

```diff
			// The full `name` will be something like "RW_24 - Kayava". We only want to announce the part after the dash.
			songToAnnounce = Regex.Split(name, " - ")[1];
-			announceAttempts = 500; // 500 attempts
+			if (MusicAnnouncementsConfig.inGameText.Value) // Gameplay announcements are enabled.
+			{
+				announceAttempts = 500; // 500 attempts
+			}
+			else
+			{
+				Debug.Log("(MusicAnnouncements) Skipping gameplay announcement due to config");
+			}
		}
```